library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -355,6 +357,7 @@ void delete() {
         for (DownloadFile downloadFile : downloadFiles) {
             downloadFile.delete();
         }
+        deleteDownloadDirs();
```

Comment:
Deleting the files and then deleting the directories is probably overkill. You can just delete the directories because it will grab all of the files inside a directory and attempt to delete them.

Reply:
Ah wait, this isn't possible because we need to update the DB persistence layer 🤔

Reply:
Not really. It still need to go through same level of effort - A directory cannot be deleted if it has even a single file/directory inside.

Reply:
This has issues if the file is downloading and you decide to delete it 😄 if we are still downloading then it will try to write to a file that no longer exists or is in the process of being deleted, which will result in an error on the client side. 

![error_when_writing](https://user-images.githubusercontent.com/3380092/63442812-a311c780-c42b-11e9-82ee-f45bb0fd1a6e.png)

Reply:
Getting this error will be hit and miss because it will be a race condition.

Reply:
I see. However, this is the not handled by the sdk originally and not issue with this PR. I shall liaise with @zegnus .

Reply:
I don't understand your comment. We wouldn't be getting this error before because of how the states operate. Can you please elaborate.

Reply:
Oh, I know what you meant. Ignore my earlier comment. Need to check the download state before deleting.

Reply:
Fixed.